File: C++/SingleInheritence.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int i, j;

        void fun()
        {
            cout<<"Inside Base Fun\n";
        }
};

class Derived : public Base
{
    public :
        int x;

        void gun()
        {
            cout<<"Inside Derived Gun\n";
        }
};

int main()
{
    Base bObj;
    Derived dObj;

    cout<<"Size of base class object : "<<sizeof(bObj)<<"\n";

    cout<<"Size of derived class object : "<<sizeof(dObj)<<"\n";


    return 0;
}

File: Java/SingleInheritence.java
class Base                      //8
{
    public int i,j;

    public void fun()
    {
        System.out.println("Inside Base Fun");
    }
}

class Derived extends Base      //12
{
    public int x;

    public void gun()
    {
        System.out.println("Inside Derived Gun");
    }
}

class SingleInheritence
{
    public static void main(String A[])
    {
        Base bObj = new Base();
        Derived dObj = new Derived();

        bObj.fun();
        dObj.fun();
        dObj.gun();

        bObj = null;
        dObj = null;

        System.gc();
    }
}